src/Widget/Attribute/FieldAttributes.php

```diff
@@ -18,53 +17,199 @@
-    private string $attribute = '';
-    private array $attributes = [];
     private string $containerClass = '';
+    /** @psalm-var array<string, string> */
+    private array $containersClass = [];
     private string $errorClass = '';
+    /** @psalm-var array<string, string> */
+    private array $errorsClass = [];
     private string $errorMessage = '';
     private string $hintClass = '';
+    /** @psalm-var array<string, string> */
+    private array $hintsClass = [];
     private string $id = '';
     private string $inputClass = '';
+    private array $inputsClass = [];
     private string $labelClass = '';
+    /** @psalm-var array<string, string> */
+    private array $labelsClass = [];
     private string $invalidClass = '';
+    /** @psalm-var array<string, string> */
+    private array $invalidsClass = [];
     private string $validClass = '';
+    /** @psalm-var array<string, string> */
+    private array $validsClass = [];
     private array $parts = [];
     private string $template = "{label}\n{input}\n{hint}\n{error}";
+    /** @psalm-var array<string, string> */
+    private array $templates = [];
+    private string $type = '';
     private string $validationStateOn = 'input';
     private ?FormModelInterface $formModel = null;
 
     /**
-     * Set aria-describedby attribute.
+     * Set container class each for field type.
+     *
+     * @param array $containerClass the container class to be used to layout the field.
+     *
+     * ```php
+     * [Field::TYPE_TEXT => 'test-class-1', Field::TYPE_SUBMIT_BUTTON => 'test-class-2']
      *
      * @return static
      *
-     * @link https://www.w3.org/TR/WCAG20-TECHS/ARIA1.html
+     * @psalm-param array<string, string> $containerClass
      */
-    public function ariaDescribedBy(): self
+    public function addContainerClass(array $containerClass): self
     {
         $new = clone $this;
-        $new->ariaDescribedBy = true;
+        $new->containersClass = $containerClass;
+        return $new;
+    }
+
+    /**
+     * Set error class each for field type.
+     *
+     * @param array $errorClass the error class to be used to layout the field.
+     *
+     * ```php
+     * [Field::TYPE_TEXT => 'test-class-1', Field::TYPE_SUBMIT_BUTTON => 'test-class-2']
+     *
+     * @return static
+     *
+     * @psalm-param array<string, string> $errorClass
+     */
+    public function addErrorClass(array $errorClass): self
+    {
+        $new = clone $this;
+        $new->errorsClass = $errorClass;
+        return $new;
+    }
+
+    /**
+     * Set hint class each for field type.
+     *
+     * @param array $hintClass the hint class to be used to layout the field.
+     *
+     * ```php
+     * [Field::TYPE_TEXT => 'test-class-1', Field::TYPE_SUBMIT_BUTTON => 'test-class-2']
+     *
+     * @return static
+     *
+     * @psalm-param array<string, string> $hintClass
+     */
+    public function addHintClass(array $hintClass): self
```

Comment:
Since it's not adding a class, it's better to name it `with*`.

```suggestion
    public function withHintClass(array $hintClass): self
```

Reply:
done.